dwave/optimization/include/dwave-optimization/nodes/interpolation.hpp

```diff
@@ -0,0 +1,65 @@
+// Copyright 2024 D-Wave
+//
+//    Licensed under the Apache License, Version 2.0 (the "License");
+//    you may not use this file except in compliance with the License.
+//    You may obtain a copy of the License at
+//
+//        http://www.apache.org/licenses/LICENSE-2.0
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+
+/// Interpolation routines.
+
+#pragma once
+
+#include <span>
+#include <vector>
+
+#include "dwave-optimization/array.hpp"
+#include "dwave-optimization/graph.hpp"
+#include "dwave-optimization/state.hpp"
+
+namespace dwave::optimization {
+class BSplineNode : public ArrayOutputMixin<ArrayNode> {
+ public:
+    explicit BSplineNode(ArrayNode* array_ptr,
+                         const int k,
+                         const std::vector<double> t,
+                         const std::vector<double> c);
+
+    double const* buff(const State& state) const override;
+    void commit(State& state) const override;
+    std::span<const Update> diff(const State&) const override;
+    void initialize_state(State& state) const override;
+    bool integral() const override;
+
+    /// @copydoc Array::minmax()
+    std::pair<double, double> minmax(
+            optional_cache_type<std::pair<double, double>> cache = std::nullopt) const override;
+
+    void propagate(State& state) const override;
+    void revert(State& state) const override;
+
+    int get_k() const;
+    const std::vector<double>& get_t() const;
+    const std::vector<double>& get_c() const;
```

Comment:
I would just call these `k()`, `t()`, and `c()`. Though they do need docstrings.